### PR TITLE
Configurable Prometheus Retention

### PR DESCRIPTION
**Pull request by patrickbrophy**
This PR addresses issue #1640. This PR adds a new configuration called `Retention`. This PR defines the default retention time as 15 days.

### CODE REVIEW

File: docs/parameters.yaml [hunk 1]
```diff
@@ -2255,6 +2255,13 @@ type: bool
 default: true
 components: ["origin", "cache", "director", "registry"]
 ---
+name: Monitoring.Retention
+description: |+
+  The duration of which Prometheus should retain the monitoring data.
+type: string
```
Comment (CannonLock) [inline docs/parameters.yaml:2261]:
```suggestion
type: duration
```

**Review by CannonLock — CHANGES_REQUESTED** [review]
Can't commit on the soundness except for how it relates to the UI, but this tweak makes sure the duration UI is used in the config rather then the default string UI.

File: docs/parameters.yaml (hunk 1 above)
Comment (patrickbrophy) [reply docs/parameters.yaml:2261]: I am not opposed to this, however using the duration type requires that the units are in the following: "ns", "us" (or "µs"), "ms", "s", "m", "h".

Comment (CannonLock) [reply docs/parameters.yaml:2261]: Oh gosh, prometheus extends time.Duration units. 

Let's either add a new type that defines this new `Prometheus Duration` or set it to `Duration`. I want to be able to check all inputs in the config UI and setting them as string prevents that. 

I slightly prefer just using duration for now, and if admins come back and say that it sucks to convert a month into hours then we can create a new type.

Comment (patrickbrophy) [reply docs/parameters.yaml:2261]: Ok for now I'll just change it to `duration`.

**Review by CannonLock — APPROVED** [review]
Syntax LGTM

File: docs/parameters.yaml [hunk 2]
```diff
@@ -2255,6 +2255,13 @@ type: bool
 default: true
 components: ["origin", "cache", "director", "registry"]
 ---
+name: Monitoring.Retention
```
Comment (jhiemstrawisc) [inline docs/parameters.yaml:2258]: Can we make the variable a little more self descriptive? What about something like "Monitoring.DataRetention" or "Monitoring.TimeSeriesRetention". Those are a bit on the long side, but I like that they help describe more about what's being retained.

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]
One small change to request a user-facing config name, otherwise looks straight forward and correct

**Review by jhiemstrawisc — APPROVED** [review]
LGTM

**Pull request merged**